src/lib.rs

```diff
@@ -94,3 +94,82 @@ mod tests {
         assert_eq!(p, expected);
     }
 }
+
+/// A safe wrapper for a path with only a single component.
+/// This prevents path traversal attacks.
+///
+/// It just allows a single normal path element and no parent, root directory or prefix like `C:`.
+#[derive(Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct PathComponent {
```

Comment:
I am not happy with my naming of it. I've used it because of https://doc.rust-lang.org/std/path/enum.Component.html but it doesn't match well.
Maybe `SinglePath`, `PathElement` or even `SinglePathElement`.

Reply:
`SinglePathComponent` or `SinglePathElement` would be fine.

Reply:
Let's also mark the type `#[doc(hidden)]`. It needs to be public for use in the macro, but we can consider it an implementation detail and discourage its direct use through the `pathbuf` crate.